samples/snippets/type_system_test.py

```diff
@@ -0,0 +1,357 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pandas.testing
+
+from bigframes import dtypes
+
+
+def test_type_system_examples() -> None:
+    # [START bigquery_dataframes_type_sytem_timestamp_local_type_conversion]
+    import pandas as pd
+
+    import bigframes.pandas as bpd
+
+    s = pd.Series([pd.Timestamp("20250101")])
+    assert s.dtype == "datetime64[ns]"
+    assert bpd.read_pandas(s).dtype == "timestamp[us][pyarrow]"
+    # [END bigquery_dataframes_type_sytem_timestamp_local_type_conversion]
+
+    # [START bigquery_dataframes_type_system_pyarrow_preference]
+    import datetime
+
+    import pandas as pd
+
+    import bigframes.pandas as bpd
+
+    s = pd.Series([datetime.date(2025, 1, 1)])
+    s + pd.Timedelta(hours=12)
+    # 0	2025-01-01
+    # dtype: object
+
+    bpd.read_pandas(s) + pd.Timedelta(hours=12)
+    # 0    2025-01-01 12:00:00
+    # dtype: timestamp[us][pyarrow]
+    # [END bigquery_dataframes_type_system_pyarrow_preference]
+    pandas.testing.assert_series_equal(
+        s + pd.Timedelta(hours=12), pd.Series([datetime.date(2025, 1, 1)])
+    )
+    pandas.testing.assert_series_equal(
+        (bpd.read_pandas(s) + pd.Timedelta(hours=12)).to_pandas(),
+        pd.Series([pd.Timestamp(2025, 1, 1, 12)], dtype=dtypes.DATETIME_DTYPE),
+        check_index_type=False,
+    )
+
+    # [START bigquery_dataframes_type_system_simple_json]
+    import db_dtypes
+    import pandas as pd
+
+    import bigframes.pandas as bpd
+
+    json_data = [
+        "1",
+        '"str"',
+        "false",
+        '["a",{"b":1},null]',
+        '{"a":{"b":[1,2,3],"c":true}}',
+        None,
+    ]
+    # db_dtypes.JSONArrowType() might be changed to pa.json_(pa.string()) in the future
```

Comment:
Let's rephrase to make clear that we intend to support both for at least some time.

```suggestion
    # In addition to db_dtypes.JSONArrowType(), support for pa.json_(pa.string())
    # will be added once the ArrowDtype supports this extension type in pandas 3.0.0.
    # See: https://github.com/pandas-dev/pandas/pull/61103
```

CC @chelsea-lin I wonder if there's anything we can do in the meantime to prepare to support this? Maybe it just works already because of our struct and array of JSON support?